### ISSUE TITLE
Add support for BinaryIntegerLiteral and OctalIntegerLiteral

### ISSUE THREAD

**Issue by rwaldron** [jshint/jshint#1531]
Under the `esnext: true` option: 

```
BinaryIntegerLiteral ::
  0b BinaryDigits
  0B BinaryDigits

BinaryDigits ::
  BinaryDigit
  BinaryDigits BinaryDigit

BinaryDigit :: one of
  0 1

OctalIntegerLiteral ::
  0o OctalDigits
  0O OctalDigits

OctalDigits ::
  OctalDigit
  OctalDigits OctalDigit

OctalDigit :: one of
  0 1 2 3 4 5 6 7
```

**Comment by nicolo-ribaudo** [comment]
This issue can be closed (fixed in 56f285a5ebee4781a3a61343ede6b97a5cefb7cd)

**Comment by jugglinmike** [comment]
Indeed!
